### PR TITLE
Feat(climate-tool): Forecast page placeholder

### LINKED ISSUE #277: feat(climate-tool): forecasts page
**Is your feature request related to a problem? Please describe.**
The app will eventually show weather forecast information from met offices in the form of PDF documents. Whilst awaiting for the live data integration it would be good to start preparing some placeholder frontend components that may later be used to manage.

**Describe the solution you'd like**
- Create a new standalone page/component to display at `/climate/forecasts`
- Create placeholder components to show a list of available forecasts (e.g. mat-card elements for `annual` and `downscaled` forecasts)
- Integrate the pdf reader to display forecast pdf files

**Additional context**
Example PDF documents can be found at: https://dashboard.picsa.app/climate/forecast
While testing you may just want to put a copy of one or two of the pdf files in the local assets folder (can work on server sync later)

See the manual tool for example embedding a pdf reader

### PR DESCRIPTION
**Pull request by khalifan-kfan**
## Description

This pull request adds the page for climate forecast in the climate tool.

## Discussion

closes #277

## Screenshots / Videos

Please check the route  ```localhost:4200/climate/forecast```  to test.

Screenshot 2024-05-29 at 22 37 36

on click (dummy pdf)
Screenshot 2024-05-29 at 22 37 27

### CODE REVIEW

**Review by chrismclarke — CHANGES_REQUESTED** [review]
Thanks @khalifan-kfan 
From what I can see it's looking in a good state (will just need to hook up the live data when available and do some minor tidying), although I noticed this PR also has the partial changes from the resources share PR which still needs a bit of work. 

So either it would be better to wait until after that is merged, or [cherry-pick](https://docs.github.com/en/desktop/managing-commits/cherry-picking-a-commit-in-github-desktop) the relevant commits for the forecasts feature and put them on a clean branch and open a new pr.

Let me know what approach you would like to take and I can help assist

**Comment by khalifan-kfan** [comment]
Ohh sorry about that, let me first attempt the cherry-picking and see if it could help

**Review by chrismclarke — APPROVED** [review]
Thanks @khalifan-kfan 

All looks good now, thanks for tidying up the commits. 

I've added one small commit [3c10a17](https://github.com/e-picsa/picsa-apps/pull/281/commits/3c10a17fe5d6319d9dd1565173d6f8dbb7031ea4) to refactor some `*ngIf` to `@if(...)` syntax and replace the module with standalone component just because I'm slowly trying to migrate more of the code base to use the modern syntax. I've also fixed an issue where the slide-in animation wasn't showing (basically the parent div needs to always be available with a `.active` class selector and child content contain the if condition instead)

I've also removed the placeholder forecast for now until we get the feature fully integrated with the climate api, but happy to merge as-is and then add a new issue to complete the feature when ready.

**Pull request merged**
